roboteam_ai/src/skills/Keeper.cpp

```diff
@@ -127,5 +164,57 @@ Vector2 Keeper::computeBlockPoint(Vector2 defendPos) {
     return blockPos;
 }
 
+Vector2 Keeper::computeBlockPointWithAttacker(shared_ptr<roboteam_msgs::WorldRobot> attacker) {
+    Vector2 interceptionPoint;
+
+    Vector2 start;
+    Vector2 end;
+    double distanceToGoal = ((Vector2)attacker->pos - Field::get_our_goal_center()).length();
+
+    start = attacker->pos;
+    end = start + (Vector2){distanceToGoal, 0}.rotate(attacker->angle);
+
+    Arc keeperCircle = control::ControlUtils::createKeeperArc();
```

Comment:
this exact object exists within the keeper class already so you do not have to recreate it everytime. (e.g. just use blockCircle here)